### PR TITLE
Merge ServicePoint assembly into Requests

### PR DESCRIPTION
**Pull request by liveans**
Preparation for `HttpWebRequest`, `ServicePoint`, and `ServicePointManager`'s missing parts implementation.

### CODE REVIEW

File: src/libraries/shims/System.Net.ServicePoint/src/System.Net.ServicePoint.csproj [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
```
Comment (ViktorHofer) [inline src/libraries/shims/System.Net.ServicePoint/src/System.Net.ServicePoint.csproj:2]: nit: add empty lines before and after the project tag (L2 & L9).

File: src/libraries/shims/System.Net.ServicePoint/src/System.Net.ServicePoint.csproj [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+    <StrongNameKeyId>Microsoft</StrongNameKeyId>
```
Comment (ViktorHofer) [inline src/libraries/shims/System.Net.ServicePoint/src/System.Net.ServicePoint.csproj:3]: You must not change the public key token of the assembly. It was "Open" before. As open is already the default, just remove the property:

```suggestion
```

File: src/libraries/System.Net.Requests/src/System/Net/ServicePoint/BindIPEndPoint.cs [hunk 3]
```diff
@@ -3,5 +3,6 @@
 
 namespace System.Net
 {
+    [System.Runtime.CompilerServices.TypeForwardedFrom("System.Net.ServicePoint, Version=8.0.0.0, Culture=neutral, PublicKeyToken=cc7b13ffcd2ddd51")]
```
Comment (ViktorHofer) [inline src/libraries/System.Net.Requests/src/System/Net/ServicePoint/BindIPEndPoint.cs:6]: This shouldn't be necessary. TypeForwardedFrom attributes are only necessary for binary serialization which these types don't support.

**Review by ViktorHofer — COMMENTED** [review]
Please make sure that there aren't any any other hits of "System.Net.ServicePoint", i.e. in tests.proj.

File: src/libraries/System.Net.Requests/ref/System.Net.Requests.cs [hunk 4]
```diff
@@ -453,6 +453,66 @@ public abstract partial class WebResponse : System.MarshalByRefObject, System.ID
...
+
+    public delegate System.Net.IPEndPoint BindIPEndPoint(System.Net.ServicePoint servicePoint, System.Net.IPEndPoint remoteEndPoint, int retryCount);
+    [System.FlagsAttribute]
+    public enum SecurityProtocolType
+    {
+        SystemDefault = 0,
+        [System.ObsoleteAttribute("SecurityProtocolType.Ssl3 has been deprecated and is not supported.")]
+        Ssl3 = 48,
+        Tls = 192,
+        Tls11 = 768,
```
[the remark this answers is not in the document]
Comment (ViktorHofer) [reply src/libraries/System.Net.Requests/ref/System.Net.Requests.cs:465]: This should definitely happen in a different PR to keep this scoped to the merge.

**Review by ViktorHofer — APPROVED** [review]

File: src/libraries/System.Net.Requests/ref/System.Net.Requests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (liveans) [reply src/libraries/System.Net.Requests/ref/System.Net.Requests.cs:464]: I'll address this as a separate PR

**Review by wfurt — APPROVED** [review]
LGTM....but I'm not the expert. I would assume issues would surface via API compat or test failures.

**Comment by ViktorHofer** [comment]
Changes look good and APICompat ran successfully. Ready to go.

**Comment by liveans** [comment]
CI Failures unrelated

**Pull request merged**